File: code/graphs/graph-adt.py
class Graph:
    def __init__(self, n):
        self.V = n
        self.adj = [[] for _ in range(n)]   # adj[u] = u's neighbors

    def add_edge(self, u, v):
        self.adj[u].append(v)
        self.adj[v].append(u)               # omit this line for a digraph

File: code/graphs/graph-adt.rs
struct Graph {
    adj: Vec<Vec<usize>>,               // adj[u] = u's neighbors
}
impl Graph {
    fn new(n: usize) -> Self {
        Graph { adj: vec![Vec::new(); n] }
    }
    fn add_edge(&mut self, u: usize, v: usize) {
        self.adj[u].push(v);
        self.adj[v].push(u);            // omit this line for a digraph
    }
}
